pkg/framework/webhooks.go

```diff
@@ -0,0 +1,149 @@
+var DefaultValidatingWebhookConfiguration = mapiv1.NewValidatingWebhookConfiguration()
+
+// DefaultMutatingWebhookConfiguration is a default mutating webhook configuration resource provided by MAO
+var DefaultMutatingWebhookConfiguration = mapiv1.NewMutatingWebhookConfiguration()
+
+// GetMutatingWebhookConfiguration gets MutatingWebhookConfiguration object by name
+func GetMutatingWebhookConfiguration(c client.Client, name string) (*admissionregistrationv1.MutatingWebhookConfiguration, error) {
+	key := client.ObjectKey{Name: name}
+	existing := &admissionregistrationv1.MutatingWebhookConfiguration{}
+
+	if err := wait.PollImmediate(RetryShort, WaitShort, func() (bool, error) {
+		if err := c.Get(context.TODO(), key, existing); err != nil {
+			klog.Errorf("Error querying api for MutatingWebhookConfiguration object %q: %v, retrying...", name, err)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		return nil, fmt.Errorf("error getting MutatingWebhookConfiguration %q: %v", name, err)
+	}
+	return existing, nil
+}
+
+// GetValidatingWebhookConfiguration gets ValidatingWebhookConfiguration object by name
+func GetValidatingWebhookConfiguration(c client.Client, name string) (*admissionregistrationv1.ValidatingWebhookConfiguration, error) {
+	key := client.ObjectKey{Name: name}
+	existing := &admissionregistrationv1.ValidatingWebhookConfiguration{}
+
+	if err := wait.PollImmediate(RetryShort, WaitShort, func() (bool, error) {
+		if err := c.Get(context.TODO(), key, existing); err != nil {
+			klog.Errorf("Error querying api for ValidatingWebhookConfiguration object %q: %v, retrying...", name, err)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		return nil, fmt.Errorf("error getting ValidatingWebhookConfiguration %q: %v", name, err)
+	}
+	return existing, nil
+}
+
+// DeleteValidatingWebhookConfiguration deletes the specified ValidatingWebhookConfiguration object
+func DeleteValidatingWebhookConfiguration(c client.Client, webhookConfiguraiton *admissionregistrationv1.ValidatingWebhookConfiguration) error {
+	return wait.PollImmediate(RetryShort, WaitShort, func() (bool, error) {
+		if err := c.Delete(context.TODO(), webhookConfiguraiton); err != nil {
+			klog.Errorf("error querying api for ValidatingWebhookConfiguration object %q: %v, retrying...", webhookConfiguraiton.Name, err)
+			return false, nil
+		}
+		return true, nil
+	})
+}
+
+// DeleteMutatingWebhookConfiguration deletes the specified MutatingWebhookConfiguration object
+func DeleteMutatingWebhookConfiguration(c client.Client, webhookConfiguraiton *admissionregistrationv1.MutatingWebhookConfiguration) error {
+	return wait.PollImmediate(RetryShort, WaitShort, func() (bool, error) {
+		if err := c.Delete(context.TODO(), webhookConfiguraiton); err != nil {
+			klog.Errorf("error querying api for MutatingWebhookConfiguration object %q: %v, retrying...", webhookConfiguraiton.Name, err)
+			return false, nil
+		}
+		return true, nil
```

Comment:
What about trying to delete a resource that doesn't exist? Should we account for that?